metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -78,10 +78,56 @@ type enricher struct {
 
 const selector = "kubernetes"
 
+const (
+	PodResource                   = "pod"
+	ServiceResource               = "service"
+	DeploymentResource            = "deployment"
+	ReplicaSetResource            = "replicaset"
+	StatefulSetResource           = "statefulset"
+	DaemonSetResource             = "daemonset"
+	JobResource                   = "job"
+	NodeResource                  = "node"
+	CronJobResource               = "cronjob"
+	PersistentVolumeResource      = "persistentvolume"
+	PersistentVolumeClaimResource = "persistentvolumeclaim"
+	StorageClassResource          = "storageclass"
+)
+
+func getResource(resourceName string) kubernetes.Resource {
```

Comment:
Why do we need both the constants and this function ? Why not just setting PodResource to `&kubernetes.Pod{}` instead of a string ?
I am trying to understand the benefit for the extra code

Reply:
Using a string we can pass the name as a constant and we don't have to worry about the resource separately. See `init()` function for this use

Reply:
Ok thanks. Leave it as it is